benches/basic.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use bytes::Bytes;
+use criterion::{criterion_group, criterion_main, Criterion};
+use pprof::criterion::{Output, PProfProfiler};
+
+use ztunnel::state::ProxyState;
+use ztunnel::strng;
+use ztunnel::xds::ProxyStateUpdateMutator;
+
+pub fn xds(c: &mut Criterion) {
+    use ztunnel::xds::istio::workload::Port;
+    use ztunnel::xds::istio::workload::Service as XdsService;
+    use ztunnel::xds::istio::workload::Workload as XdsWorkload;
+    use ztunnel::xds::istio::workload::{NetworkAddress as XdsNetworkAddress, PortList};
+    let mut c = c.benchmark_group("xds");
+    // let updater = ProxyStateUpdater::new(state.clone(), Arc::new(ztunnel::cert_fetcher::NoCertFetcher()));
+    let mut state = ProxyState::default();
+    let updater = ProxyStateUpdateMutator::new_no_fetch();
+    let svc = XdsService {
+        hostname: "example.com".to_string(),
+        addresses: vec![XdsNetworkAddress {
+            network: "".to_string(),
+            address: vec![127, 0, 0, 3],
+        }],
+        ..Default::default()
+    };
+    updater.insert_service(&mut state, svc).unwrap();
```

Comment:
all this is done in the bench function as well. was the intent to capture these as a closure and reuse them?

Reply:
Thanks! forgot to clean this up